clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -744,34 +798,17 @@ public void testFencedMemberStopHeartbeatUntilItReleasesAssignmentToRejoin() {
         verify(heartbeatRequestState).onFailedAttempt(anyLong());
         verify(heartbeatRequestState).reset();
 
+        when(heartbeatRequestState.canSendRequest(anyLong())).thenReturn(false);
         when(membershipManager.state()).thenReturn(MemberState.FENCED);
```

Comment:
what the HB manager checks on the membershipMgr in this case is the shouldSKip, so I guess what we need to do here is to mock the fencing is: `when(membershipManager.shouldSkipHeartbeat()).thenReturn(true);` (wonder if this will allow you to loose the when canSendRequest too, which I would expect should work because the heartbeatRequestState is not a mock)